apis/fluent/atrium-api-fluent/src/commonMain/kotlin/ch/tutteli/atrium/api/fluent/en_GB/anyExpectations.kt

```diff
@@ -167,6 +167,14 @@ internal fun <TSub : Any> Expect<*>.toBeAnInstanceOf(kClass: KClass<TSub>): Subj
 inline fun <reified TSub : Any> Expect<*>.toBeAnInstanceOf(noinline assertionCreator: Expect<TSub>.() -> Unit): Expect<TSub> =
     toBeAnInstanceOf(TSub::class).transformAndAppend(assertionCreator)
 
+inline fun <reified TSub : Any> Expect<*>.notToBeAnInstanceOf(): Expect<Boolean> = notToBeAnInstanceOf(TSub::class).transform()
```

Comment:
IMO we should not transform to Boolean but keep the same subject as before. I would imagine that someone is using it like the following:
```
sealed interface A {
  val someInt: Int
}
class B: A { ... }
class C: A { ... }
class D: A { ... }

expect(someA).because("can be any of A but not D because ...") {
  notToBeAnInstanceOf<D>()
    .feature(A::someInt)  // extract feature of A, would not be possible if we transform to Boolean
    .toBeEqualTo(5)
}
```

<hr/>

please add a KDOC and link to your sample (see other functions in this file as guidance)

Reply:
This is actually quite interesting, I just now realize it is possible to chain tests without regards to one another. Thanks for sharing this knowledge! But just one question : Won't this make the test coupled? If say, for instance, we sometime along the design realize that we have to change the implementation of class A (due to some random reason). I'll do the changes you requested, but I'm just curios of this aspect!

Reply:
not sure I get what you mean by coupled. The above example is of course fictional but if it is true that for some (business) case all implementations (maybe expect for D) need to return 5 for `someInt` then I don't see coupling here but merely testing the behaviour of the interface. But maybe you meant something else?

Reply:
No, you explained it fine, was just a bit confuse about something